### PR TITLE
Added and implemented delta time

### PR DESCRIPTION
**Pull request by GarrettBassen**
# Description

Major engine changes and reworks to allow for similar play on different devices. Movement and animations are now time-dependent using delta time as opposed to frame-dependent. This allows the game to run the same without need for locking frame rate with former sync() function or vsync. Fixed update was added which is updated 60 times per second regardless of frame rate.

# Fixes and Changes

Timer.java:
Removed unused and redundant functions and variables. Added delta time to make calculations time dependent instead of frame dependent. Made functions static

Player.Java:
Removed redundant import
Changed variable speeds
Changed delta position directions
Added delta time in movement to make player movement time-dependent instead of frame-dependent

Animation.java:
Formatted file
Removed redundant and unused variables
Renamed variables and functions for readability
Fixed run() to use new delta time

GameEngine.java:
Removed redundant and unused variables
Frame unlocked update()
Added fixedUpdate()
Implemented delta time in fixed update
Removed sync() as it was redundant and overridden by glfwSwapInterval(1) Implemented delta time in FPS counter

GameObject.java:
Formatted variables

GameLogic.java:
Frame-unlocked update function

Game.java:
Changed update() to work with new changes.
Made camera follow player

Main.java:
Formatted GameEngine() arguments for readability

MapTree.java:
Small formatting

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

# Possible Bugs and Errors

Animations, while called in fixedUpdate(), are slightly slower when vsync is enabled compared to when frame rate is unlocked. Movement does not appear to be effected by this bug and there does not appear to be any impact on game play.

Tests were conducted to ensure movement speed was the same among multiple frame rates. Further testing on separate devices is encouraged.

### CODE REVIEW

File: src/main/java/com/github/jacksonhoggard/voodoo2d/engine/Timer.java [hunk 1]
```diff
@@ -2,24 +2,20 @@
...
-    
-    public void init() {
-        lastLoopTime = getTime();
-    }
+    private static double lastLoopTime;
+    private static double deltaTime;
 
     public static double getTime() {
-        return System.nanoTime() / 1000_000_000.0;
+        return System.nanoTime() / 1000000000f; // Time in seconds
```
Comment (JacksonHoggard) [inline src/main/java/com/github/jacksonhoggard/voodoo2d/engine/Timer.java]:
```suggestion
        return System.nanoTime() / 1000000000D; // Time in seconds
```

File: src/main/java/com/github/jacksonhoggard/voodoo2d/engine/GameEngine.java [hunk 2]
```diff
@@ -44,34 +40,24 @@ public void run() {
...
+            update();
             render();
 
-            if (!window.isvSync()) {
-                sync();
+            // Fixed update gets called 60 times a second
+            time += Timer.getDeltaTime();
+            if (time > 0.016) {
+                fixedUpdate();
+                time = 0;
```
Comment (JacksonHoggard) [inline src/main/java/com/github/jacksonhoggard/voodoo2d/engine/GameEngine.java:56]: fixedUpdate should be called before update. This seems to fix the stuttering animations and brings the frame rate up to 60.

**Review by JacksonHoggard — CHANGES_REQUESTED** [review]
A very welcome change to the engine, the fixedUpdate method will be especially useful later on. Check out the comments I made and make any necessary changes and then I'll merge the changes. Thanks!

**Comment by GarrettBassen** [comment]
Changes were made. Happy thanksgiving!

**Review by JacksonHoggard — APPROVED** [review]
Nice work! I'm grateful for all of your contributions recently. Happy Thanksgiving!

**Pull request merged**